gui/options.cpp

```diff
@@ -1878,39 +1878,60 @@ void GlobalOptionsDialog::apply() {
 #endif // USE_SDL_NET
 #endif // USE_CLOUD
 
-	if (!_newTheme.empty())
+	Common::String oldThemeId = g_gui.theme()->getThemeId();
+	Common::String oldThemeName = g_gui.theme()->getThemeName();
+	if (!_newTheme.empty()) {
+		ConfMan.set("gui_theme", _newTheme);
 		redraw = true;
+	}
 
 #ifdef USE_TRANSLATION
-	int selLang = _guiLanguagePopUp->getSelectedTag();
+	int selectedLang = _guiLanguagePopUp->getSelectedTag();
 	Common::String oldLang = ConfMan.get("gui_language");
-	ConfMan.set("gui_language", TransMan.getLangById(selLang));
-	Common::String newLang = ConfMan.get("gui_language");
-
+	Common::String newLang = TransMan.getLangById(selectedLang);
+	Common::String newCharset;
 	if (newLang != oldLang) {
 		TransMan.setLanguage(newLang);
+		ConfMan.set("gui_language", newLang);
+		newCharset = TransMan.getCurrentCharset();
 		redraw = true;
 	}
 #endif
 
 	GUI::ThemeEngine::GraphicsMode gfxMode = (GUI::ThemeEngine::GraphicsMode)_rendererPopUp->getSelectedTag();
-	const char *cfg = GUI::ThemeEngine::findModeConfigName(gfxMode);
-	if (!ConfMan.get("gui_renderer").equalsIgnoreCase(cfg)) {
-		ConfMan.set("gui_renderer", cfg, _domain);
+	Common::String oldGfxConfig = ConfMan.get("gui_renderer");
+	Common::String newGfxConfig = GUI::ThemeEngine::findModeConfigName(gfxMode);
+	if (newGfxConfig != oldGfxConfig) {
```

Comment:
Is there any reason here to change the case-insensitive test into a case-sensitive one?